### PR TITLE
Upgrade to sphinx 8

### PR DESCRIPTION
**Pull request by agoscinski**
I wanted to use `linkcheck_anchors_ignore_for_url` which is available sphinx>=0.7.1. I thought I just update directly to sphinx 8.

### CODE REVIEW

**Comment by agoscinski** [comment]
I am not sure what to do with this not working link, seems rather important, maybe their page has a hicks

**Comment by GeigerJ2** [comment]
I just stumbled on the CI failing for my latest blog-post PR #72. Weird, because this wasn't an issue for the other one (PR #69). So if there are no arguments against the updates here, I'd say LGTM. Any objections, @edan-bainglass, could you look into it?

File: requirements.txt [hunk 1]
```diff
@@ -1,7 +1,9 @@
-myst-parser~=0.18.0
-pydata-sphinx-theme==0.15.2
+sphinx~=8.0.0
+myst-parser~=4.0.0
```
Comment (GeigerJ2) [inline requirements.txt:2]: Crazy that we're jumping 4 major versions here :D I guess this wasn't updated in a while before.

File: docs/conf.py [hunk 2]
```diff
@@ -29,25 +30,19 @@
...
 # see https://pydata-sphinx-theme.readthedocs.io
 html_theme_options = {
     # "announcement": "Welcome to the new AiiDA site!",
     "logo": {
         "image_light": "logo-light.svg",
         "image_dark": "logo-dark.svg",
     },
-    "favicons": [
-        {
-            "rel": "icon",
```
Comment (GeigerJ2) [inline docs/conf.py:41]: Where do these specifications go in the PR?

**Review by GeigerJ2 — APPROVED** [review]
Just two minor comments. LGTM!

File: docs/conf.py (hunk 2 above)
Comment (agoscinski) [reply docs/conf.py:41]: Is the commit message clear?
https://github.com/aiidateam/aiida-website/pull/71/commits/b51f31552d33b1fb5317928cfda59ac8b76aadd7

File: requirements.txt (hunk 1 above)
Comment (agoscinski) [reply requirements.txt:2]: or python2.7 in aiida-plugin-cutter ;)

File: docs/conf.py (hunk 2 above)
Comment (agoscinski) [reply docs/conf.py:41]: We don't have to specify, the are automatically determined, but I can also add them 

> For any attributes that you don’t explicitly set, Sphinx Favicon will infer the values from your provided input. For example: if you don’t provide a type attribute, Sphinx Favicon will infer the type from the file extension. If you don’t provide a sizes attribute, Sphinx Favicon will attempt to read the pixel dimensions of the provided image.
https://sphinx-favicon.readthedocs.io/en/stable/configuration.html#configuration

Comment (agoscinski) [reply docs/conf.py:41]: I'll just add them back to make it easier

**Pull request merged**